.github/workflows/main.yaml

```diff
@@ -0,0 +1,161 @@
+name: Quality Checks
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout the code
+      - name: Checkout
+        uses: actions/checkout@v4
+      # Install Python
```

Comment:
Nice comments, but if the comment and name is the same, is it needed?

Reply:
Comments help provide more detail and creates separation within the file to make it easier to read.